src/v2/providers/identity.ts

```diff
@@ -0,0 +1,233 @@
+import { BlockingFunction } from '../../cloud-functions';
+import {
+  AuthBlockingEvent,
+  AuthBlockingEventType,
+  BeforeCreateResponse,
+  BeforeSignInResponse,
+  wrapHandler,
+} from '../../common/providers/identity';
+import * as options from '../options';
+
+/** Internally used when parsing the options. */
+interface InternalOptions {
+  opts: options.GlobalOptions;
+  idToken: boolean;
+  accessToken: boolean;
+  refreshToken: boolean;
+}
+
+/**
+ * All function options plus idToken, accessToken, and refreshToken.
+ */
+export interface BlockingOptions extends options.GlobalOptions {
+  idToken?: boolean;
+  accessToken?: boolean;
+  refreshToken?: boolean;
+}
+
+/**
+ * Handle an event that is triggered before a user is created.
+ */
+export function beforeUserCreated(
+  handler: (
```

Comment:
nit* The # of duplication here passes my threshold for defining a type!

Reply:
Unfortunately we'd lose out on typescript type completion by making this a separate type

Reply:
wow that sucks